plugins/agents/muninpluginrunner/exec.go

```diff
@@ -0,0 +1,74 @@
+package muninpluginrunner
+
+import (
+	"bufio"
+	"regexp"
+	"strconv"
+
+	"github.com/abrander/agento/plugins"
+	"github.com/abrander/agento/timeseries"
+)
+
+func init() {
+	plugins.Register("muninpluginrunner", newMuninPluginRunner)
+}
+
+// MuninPluginRunner will retrieve stub status.
+type MuninPluginRunner struct {
+	Cmd string `toml:"cmd" json:"cmd" description:"Command to run"`
+	Arg string `toml:"arg" json:"arg" description:"Arguments to command"`
+	kv  []KeyValue
+}
+
+type KeyValue struct {
+	key   string
+	value float64
+}
+
+func newMuninPluginRunner() interface{} {
+	return new(MuninPluginRunner)
+}
+
+// Gather will execute command (with arguments) and read each line in output.
+// Gather expect output to be munin plugin style:
+// http://munin-monitoring.org/wiki/HowToWritePlugins
+func (m *MuninPluginRunner) Gather(transport plugins.Transport) error {
+	stdout, _, _ := transport.Exec(m.Cmd, m.Arg)
+
+	scanner := bufio.NewScanner(stdout)
+	for scanner.Scan() {
+		re := regexp.MustCompile("^(.*).value ([0-9]+(\\.([0-9])*)?)$")
+		matches := re.FindAllStringSubmatch(scanner.Text(), -1)
+
+		if len(matches) == 1 {
+			value, _ := strconv.ParseFloat(matches[0][2], 64)
```

Comment:
Is it by design that it will fail silently here?

Reply:
@abrander because of the regex it will not be able to fail, it will always get a value it can parse as float. But I will add returning of err here as well